pkg/controller/jobs/kubeflow/jobs/pytorchjob/pytorchjob_controller.go

```diff
@@ -85,33 +85,8 @@ func (j *JobControl) JobStatus() kftraining.JobStatus {
 	return j.Status
 }
 
-func (j *JobControl) OrderedReplicaTypes(replicaSpecs map[kftraining.ReplicaType]*kftraining.ReplicaSpec) []kftraining.ReplicaType {
-	result := make([]kftraining.ReplicaType, 0, 2)
-	if _, ok := replicaSpecs[kftraining.PyTorchJobReplicaTypeMaster]; ok {
-		result = append(result, kftraining.PyTorchJobReplicaTypeMaster)
-	}
-	if _, ok := replicaSpecs[kftraining.PyTorchJobReplicaTypeWorker]; ok {
-		result = append(result, kftraining.PyTorchJobReplicaTypeWorker)
-	}
-	return result
-}
-
-// PriorityClass calculates the priorityClass name needed for workload according to the following priorities:
-//  1. .spec.runPolicy.schedulingPolicy.priorityClass
-//  2. .spec.replicaSpecs[Master].template.spec.priorityClassName
-//  3. .spec.replicaSpecs[Worker].template.spec.priorityClassName
-//
-// This function is inspired by an analogous one in mpi-controller:
-// https://github.com/kubeflow/mpi-operator/blob/5946ef4157599a474ab82ff80e780d5c2546c9ee/pkg/controller/podgroup.go#L69-L72
-func (j *JobControl) PriorityClass() string {
-	if j.Spec.RunPolicy.SchedulingPolicy != nil && len(j.Spec.RunPolicy.SchedulingPolicy.PriorityClass) != 0 {
-		return j.Spec.RunPolicy.SchedulingPolicy.PriorityClass
-	} else if m := j.Spec.PyTorchReplicaSpecs[kftraining.PyTorchJobReplicaTypeMaster]; m != nil && len(m.Template.Spec.PriorityClassName) != 0 {
-		return m.Template.Spec.PriorityClassName
-	} else if w := j.Spec.PyTorchReplicaSpecs[kftraining.PyTorchJobReplicaTypeWorker]; w != nil && len(w.Template.Spec.PriorityClassName) != 0 {
-		return w.Template.Spec.PriorityClassName
-	}
-	return ""
+func (j *JobControl) OrderedReplicaTypes() []kftraining.ReplicaType {
+	return []kftraining.ReplicaType{kftraining.PyTorchJobReplicaTypeMaster, kftraining.PyTorchJobReplicaTypeWorker}
```

Comment:
One question here: we ordered the types is to 1) make sure the scheduling order of different types or 2) just to list the expected types here, which means we can shuffle the order, like scheduling the worker types first. 

We may want to schedule the workers first which dominates the resource allocation. Thanks!

Reply:
IIUC, the PyTorchJob replicaspec is map, which means no order between each replica type.
So, I guess that we can not enforce the scheduling order.